lib/puppet/application/describe.rb

```diff
@@ -67,13 +67,15 @@ def list_types
     @types.keys.sort_by(&:to_s).each do |name|
       type = @types[name]
       s = type.doc.gsub(/\s+/, " ")
-      n = s.index(". ")
-      if n.nil?
+      if s.empty?
         s = ".. no documentation .."
-      elsif n > 45
-        s = s[0, 45] + " ..."
       else
-        s = s[0, n]
+        n = s.index(". ")
+        if n.nil? || n > 45
+          s = s[0, 45] + " ..."
```

Comment:
If there is only one sentence in the description and it's less than 45 characters, we'll still add the ellipses to the description output, which indicates there's more to be read, which is misleading.

```
irb(main):008:0> 'documentation'.index(". ")
=> nil
irb(main):009:0> 'documentation'[0, 45]
=> "documentation"
```

If you're in here making improvements already, you should add another case to check to see if `s.size < 45` and just output all of `s` with no ellipses.

Reply:
Updated